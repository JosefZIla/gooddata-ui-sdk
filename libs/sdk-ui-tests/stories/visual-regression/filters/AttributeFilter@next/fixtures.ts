// (C) 2022 GoodData Corporation

export const ElementUris = {
    CompuSci: "/gdc/md/referenceworkspace/obj/1054/elements?id=165678",
    Educationly: "/gdc/md/referenceworkspace/obj/1054/elements?id=165847",
    Explorer: "/gdc/md/referenceworkspace/obj/1054/elements?id=342345",
    "Grammar Plus": "/gdc/md/referenceworkspace/obj/1054/elements?id=168737",
    PhoenixSoft: "/gdc/md/referenceworkspace/obj/1054/elements?id=166339",
    TouchAll: "/gdc/md/referenceworkspace/obj/1054/elements?id=460491",
    WonderKid: "/gdc/md/referenceworkspace/obj/1054/elements?id=166497",
};
