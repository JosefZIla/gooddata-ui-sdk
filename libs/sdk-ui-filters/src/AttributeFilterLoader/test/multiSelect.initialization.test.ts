// (C) 2019-2022 GoodData Corporation
import { newTestAttributeFilterHandler } from "./fixtures";
import { waitForAsync } from "./testUtils";

describe("MultiSelectAttributeFilterHandler", () => {
    it("should trigger init success callback", async () => {
        const onInicStart = jest.fn();
        const onInitSuccess = jest.fn();
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        attributeFilterHandler.onInitStart(onInicStart);
        attributeFilterHandler.onInitSuccess(onInitSuccess);

        attributeFilterHandler.init("init");

        await waitForAsync();

        expect(onInicStart).toHaveBeenCalledTimes(1);
        expect(onInicStart.mock.calls[0]).toMatchSnapshot("onInicStart parameters");
        expect(onInitSuccess).toHaveBeenCalledTimes(1);
        expect(onInitSuccess.mock.calls[0]).toMatchSnapshot("onInitSuccess parameters");
    });

    it("should trigger init error callback", async () => {
        const onInicStart = jest.fn();
        const onInitError = jest.fn();
        const attributeFilterHandler = newTestAttributeFilterHandler("nonExisting");

        attributeFilterHandler.onInitStart(onInicStart);
        attributeFilterHandler.onInitError(onInitError);

        attributeFilterHandler.init("init");

        await waitForAsync();

        expect(onInicStart).toHaveBeenCalledTimes(1);
        expect(onInicStart.mock.calls[0]).toMatchSnapshot("onInicStart parameters");
        expect(onInitError).toHaveBeenCalledTimes(1);
        expect(onInitError.mock.calls[0]).toMatchSnapshot("onInitError parameters");
    });

    it("should trigger init cancel callback on another init call", async () => {
        const onInicStart = jest.fn();
        const onInitCancel = jest.fn();
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        attributeFilterHandler.onInitStart(onInicStart);
        attributeFilterHandler.onInitCancel(onInitCancel);

        attributeFilterHandler.init("init1");
        attributeFilterHandler.init("init2");

        await waitForAsync();

        expect(onInicStart).toHaveBeenCalledTimes(2);
        expect(onInicStart.mock.calls[0]).toMatchSnapshot("onInicStart 1 parameters");
        expect(onInicStart.mock.calls[1]).toMatchSnapshot("onInicStart 2 parameters");
        expect(onInitCancel).toHaveBeenCalledTimes(1);
        expect(onInitCancel.mock.calls[1]).toMatchSnapshot("onInitCancel parameters");
    });
});
