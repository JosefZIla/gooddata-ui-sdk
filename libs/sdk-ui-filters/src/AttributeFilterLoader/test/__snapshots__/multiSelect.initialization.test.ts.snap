// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectAttributeFilterHandler should trigger init cancel callback on another init call: onInicStart 1 parameters 1`] = `
Array [
  Object {
    "correlation": "init1",
  },
]
`;

exports[`MultiSelectAttributeFilterHandler should trigger init cancel callback on another init call: onInicStart 2 parameters 1`] = `
Array [
  Object {
    "correlation": "init2",
  },
]
`;

exports[`MultiSelectAttributeFilterHandler should trigger init cancel callback on another init call: onInitCancel parameters 1`] = `undefined`;

exports[`MultiSelectAttributeFilterHandler should trigger init error callback: onInicStart parameters 1`] = `
Array [
  Object {
    "correlation": "init",
  },
]
`;

exports[`MultiSelectAttributeFilterHandler should trigger init error callback: onInitError parameters 1`] = `
Array [
  Object {
    "correlation": "init",
    "error": [Error: No attribute recording non-existing-displayForm],
  },
]
`;

exports[`MultiSelectAttributeFilterHandler should trigger init success callback: onInicStart parameters 1`] = `
Array [
  Object {
    "correlation": "init",
  },
]
`;

exports[`MultiSelectAttributeFilterHandler should trigger init success callback: onInitSuccess parameters 1`] = `
Array [
  Object {
    "correlation": "init",
  },
]
`;
